### ISSUE TITLE
🛠️ `Checkbox` story failing a11y color contrast

### ISSUE THREAD

**Issue by mnigh** [GCTC-NTGC/gc-digital-talent#17101]
### 🛠️ Platform

`packages/forms/src/components/Checkbox/Checkbox.stories.tsx`

> The color contrast between text and its background meets WCAG AA contrast ratio.

### 🕵️ Details

As part of the tailwind migration, the text colour was changed instead of the background colour.

## Screenshot

<img width="1673" height="824" alt="Image" src="https://github.com/user-attachments/assets/03073088-e769-4cb3-87b3-80d6a229e6c9" />

### 🙋‍♀️ Proposed Solution

Replace `text-error-200` with `bg-error-200 dark:bg-error-500`.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/e62c2c31301c09cc7200503827c3b7c6c1ac31fb/packages/forms/src/components/Checkbox/Checkbox.stories.tsx#L52

### ✅ Acceptance Criteria

- [ ] Checkbox story does not fail a11y in storybook
